ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+        cvp_instance_ip = get(self._hostvars, 'cvp_instance_ip')
+        cvp_instance_ips = get(self._hostvars, 'cvp_instance_ips')
+        if cvp_instance_ip is not None or cvp_instance_ips is not None:
+            cvp_on_prem = {'ips': []}
+            cvaas = {'ips': []}
+            if cvp_instance_ip is not None:
+                if "arista.io" in cvp_instance_ip:
+                    cvaas['ips'].append(cvp_instance_ip)
+                else:
+                    cvp_on_prem['ips'].append(cvp_instance_ip)
+            elif cvp_instance_ips is not None:
+                for cvp_ip in cvp_instance_ips:
+                    if "arista.io" in cvp_ip:
+                        cvaas['ips'].append(cvp_ip)
+                    else:
+                        cvp_on_prem['ips'].append(cvp_ip)
+
+            daemon_terminattr = {'cvaddrs': []}
+            if cvp_on_prem['ips']:
+                for cvp_instance_ip in cvp_on_prem['ips']:
+                    daemon_terminattr['cvaddrs'].append(f"{cvp_instance_ip}:"
+                                                        f"{get(self._hostvars, 'terminattr_ingestgrpcurl_port')}")
+                daemon_terminattr.update({'cvauth': {'method': 'key', 'key': get(self._hostvars,
+                                                                                 'cvp_ingestauth_key')}})
+            elif cvaas['ips']:
+                for cvp_instance_ip in cvaas['ips']:
+                    daemon_terminattr['cvaddrs'].append(f"{cvp_instance_ip}:443")
+                daemon_terminattr.update({'cvauth': {'method': 'token-secure',
+                                                     'token_file': get(self._hostvars, 'cvp_token_file',
+                                                                       '/tmp/cv-onboarding-token')}})
+
+            daemon_terminattr.update({'cvvrf': self._mgmt_interface_vrf})
+            daemon_terminattr.update({'smashexcludes': get(self._hostvars, 'terminattr_smashexcludes')})
+            daemon_terminattr.update({'ingestexclude': get(self._hostvars, 'terminattr_ingestexclude')})
+            daemon_terminattr.update({'disable_aaa': get(self._hostvars, 'terminattr_disable_aaa', False)})
+            return daemon_terminattr
+        return None
+
+    @cached_property
+    def vlan_internal_order(self):
+        range = get(self._internal_vlan_order, 'range')
+        return {
+            'allocation': get(self._internal_vlan_order, 'allocation'),
+            'range': {
+                'beginning': get(range, 'beginning'),
+                'ending': get(range, 'ending')
+            }
+        }
+
+    @cached_property
+    def event_monitor(self):
+        if get(self._hostvars, 'event_monitor') is True:
+            return {'enabled': 'true'}
+        return None
+
+    @cached_property
+    def event_handlers(self):
+        event_handlers = get(self._hostvars, 'event_handlers')
+        if event_handlers is not None and hasattr(event_handlers, '__iter__'):
```

Comment:
We don't have a check like this anywhere else. Why do we need it here?

Reply:
it was - `{% if event_handlers is defined and event_handlers is iterable %}`

Reply:
Ok, I think that is an early attempt at better input validation. Since we will get schemas now, I think we can skip this check. Just check if not None like everything else. Thanks